plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -75,57 +110,61 @@ func newOIDCAuthProvider(_ string, cfg map[string]string, persister restclient.A
 		return nil, fmt.Errorf("Must provide %s", cfgClientSecret)
 	}
 
-	var certAuthData []byte
-	var err error
-	if cfg[cfgCertificateAuthorityData] != "" {
-		certAuthData, err = base64.StdEncoding.DecodeString(cfg[cfgCertificateAuthorityData])
-		if err != nil {
-			return nil, err
+	client, ok := cache.getClient(issuer, clientID, clientSecret)
```

Comment:
There's just enough logic after the if statement to make that annoying. Let me see if I can split it out to a helper or something.